### PR TITLE
User logs in

### PR DESCRIPTION
**Pull request by JonathanSR**
creates login feature test
adds login session routes(new, create)
adds new, create to sessions_controller
adds login view page
adds simplecov
removes role column from users
creates roles table
creates user_roles table
adds flash message functionality

**needs styling to view page**

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -9,6 +9,18 @@
   </head>
 
   <body>
+  <div style='text-align:center'>
+<% flash.each do |type, message| %>
+    <% if type == "success" %>
+    <%= content_tag :div, sanitize(message), class:"alert-success" %>
```
Comment (AELSchauer) [inline app/views/layouts/application.html.erb]: This nested if..else statement to render flash stuff can be refactored.

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def create
+    user = User.find_by(email: session_params[:email])
+    if user && user.authenticate(session_params[:password])
+      session[:user_id] = user.id
+      redirect_to home_path
+      flash[:success] = "#{user.first_name} Has Successfully Logged In"
+    else
+      flash[:error] = "Incorrect entry"
```
Comment (cmacaulay) [inline app/controllers/sessions_controller.rb]: 👏 flash message!

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (cmacaulay) [inline app/views/layouts/application.html.erb]: Great - I think Thursday we can go back and refactor, and review test coverage that we've covered so far. In this early stage of the project, I'd like to push forward with functionality and refactor to get to learn the app we've built even better before moving on to the next phase. Does that sound good @AELSchauer @JonathanSR @alfosco ?

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  <%= f.text_field :cellphone, placeholder: "Cell Phone Number" %>
+
+  <%= f.text_field :password, placeholder: "Password"%>
+
+  <%= f.text_field :password_confirmation, placeholder: "Password Confirmation" %>
+
+  <%= f.submit "Create Account" %>
+
+<% end %>
```
Comment (cmacaulay) [inline app/views/users/new.html.erb]: 🎉

File: config/routes.rb [hunk 4]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :users, only: [:new, :create]
+  get '/home', to: 'users#show'
```
Comment (cmacaulay) [inline config/routes.rb]: 🎉

File: spec/features/user/user_can_log_in_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+feature 'user logs in' do
+  context 'appropriate login' do
+    scenario 'user logs in and is directed to their dashboard' do
+      User.create(first_name: "sal",
+                  last_name: "espinosa",
+                  email: "espinosa@aol.com",
```
Comment (cmacaulay) [inline spec/features/user/user_can_log_in_spec.rb]: @s-espinosa shout out! 🎉 😂  @aol

Comment (s-espinosa) [inline spec/features/user/user_can_log_in_spec.rb]: @cmacaulay I should have known it was only a matter of time! 😄

**Pull request merged**